scipy/interpolate/_bsplines.py

```diff
@@ -1866,11 +1866,35 @@ def _lsq_solve_qr(x, y, t, k, w):
     assert y.ndim == 2
 
     y_w = y * w[:, None]
-    A, offset, nc = _dierckx.data_matrix(x, t, k, w)
-    _dierckx.qr_reduce(A, offset, nc, y_w)         # modifies arguments in-place
-    c = _dierckx.fpback(A, nc, y_w)
-
-    return A, y_w, c
+    if not periodic:
+        A, offset, nc = _dierckx.data_matrix(x, t, k, w)
+        _dierckx.qr_reduce(A, offset, nc, y_w)         # modifies arguments in-place
+        c = _dierckx.fpback(A, nc, y_w)
+        return A, y_w, c
+    else:
+        print("periodic.begin")
+        # Ref: https://github.com/scipy/scipy/blob/596b586e25e34bd842b575bac134b4d6924c6556/scipy/interpolate/fitpack/fpperi.f#L221-L238
+        R, H1, H2, offset, nc = _dierckx.data_matrix(x, t, k, w, False, True)
+        assert(y.shape[1] == 1) # TODO: Update QR Reduce to account for y.shape[1] != 1
```

Comment:
Could you explain what is this?
y.ndim == 2, what is y.shape[1], does this differ between parametric and non-parametric case, or is it something else?